lib/galaxy/workflow/scheduling_manager.py

```diff
@@ -298,6 +302,10 @@ def __monitor(self):
                     if not self.monitor_running:
                         return
 
+                    if not self.is_workflow_handler:
+                        # This sidesteps the filter condition in poll_active_workflow_ids,
+                        # so that all workflows assigned to a given job handler get drained.
+                        workflow_scheduler_id = None
```

Comment:
This updates it? I don't see where.

Reply:
Just below, on line 309

Reply:
Looks like it just leads to [a SELECT](https://github.com/galaxyproject/galaxy/blob/b510597de9a8c2171ad1cac4098761411a2dce05/lib/galaxy/model/__init__.py#L5502)?

Reply:
Yeah, so setting the workflow_scheduler_id to None here skips the filter for the workflow_scheduler_id:
```
        if scheduler is not None:
            and_conditions.append(WorkflowInvocation.scheduler == scheduler)
```

Reply:
That's not the scheduler name though, that's the "plugin." In the db it'd need to set handler to `_default_`.

Reply:
I don't understand, `poll_active_workflow_ids ` takes two kwargs, `handler` and `scheduler`, and if they're given they are added as filters.

> That's not the scheduler name though, that's the "plugin." In the db it'd need to set handler to `_default_`.

I don't know what `workflow_scheduler_id ` is, but empirically it is not related to the handler column or `_default_`. The `handler` argument that is passed into `poll_active_workflow_ids` is `self.app.config.server_name`, which makes sense to me, and we don't want to override that.

Maybe there is some confusion here with creation and grabbing of new invocations, which does use `_default_` in the grabbing statement ?

Reply:
> This updates it? I don't see where.

Oh, did you mean that we don't update the invocation in the database ? Yeah, that's right, we just allow the job handlers to finish the invocations that are already assigned to them. We could instead set the `handler` column to `_default_` ... that way we wouldn't need to keep the thread going. That might be a better solution.